### ISSUE TITLE
PHP 8 support

### ISSUE THREAD

**Issue by enumag** [rosmanov/pecl-eio#12]
Running this under PHP 8.0.0 (stable):

```
pecl install eio
```

Getting this error:

```
running: make
/bin/sh /tmp/pear/temp/pear-build-defaultuserPfDmhI/eio-2.0.4/libtool --mode=compile cc -I. -I/tmp/pear/temp/eio -I/tmp/pear/temp/pear-build-defaultuserPfDmhI/eio-2.0.4/include -I/tmp/pear/temp/pear-build-defaultuserPfDmhI/eio-2.0.4/main -I/tmp/pear/temp/eio -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/php7  -DHAVE_CONFIG_H  -g -O2   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /tmp/pear/temp/eio/php7/php_eio.c -o php7/php_eio.lo
mkdir php7/.libs
 cc -I. -I/tmp/pear/temp/eio -I/tmp/pear/temp/pear-build-defaultuserPfDmhI/eio-2.0.4/include -I/tmp/pear/temp/pear-build-defaultuserPfDmhI/eio-2.0.4/main -I/tmp/pear/temp/eio -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/php7 -DHAVE_CONFIG_H -g -O2 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /tmp/pear/temp/eio/php7/php_eio.c  -fPIC -DPIC -o php7/.libs/php_eio.o
/tmp/pear/temp/eio/php7/php_eio.c: In function 'php_eio_call_method':
/tmp/pear/temp/eio/php7/php_eio.c:153:5: error: 'zend_fcall_info' {aka 'struct _zend_fcall_info'} has no member named 'no_separation'
  153 |  fci.no_separation = 1;
      |     ^
/tmp/pear/temp/eio/php7/php_eio.c: In function 'php_eio_import_func_info':
/tmp/pear/temp/eio/php7/php_eio.c:240:39: error: 'IS_CALLABLE_STRICT' undeclared (first use in this function)
  240 |   if (!zend_is_callable_ex(zcb, NULL, IS_CALLABLE_STRICT, NULL, &fcc, &error)) {
      |                                       ^~~~~~~~~~~~~~~~~~
/tmp/pear/temp/eio/php7/php_eio.c:240:39: note: each undeclared identifier is reported only once for each function it appears in
/tmp/pear/temp/eio/php7/php_eio.c: In function 'php_eio_new_eio_cb':
/tmp/pear/temp/eio/php7/php_eio.c:496:2: warning: implicit declaration of function 'TSRMLS_SET_CTX' [-Wimplicit-function-declaration]
  496 |  TSRMLS_SET_CTX(eio_cb->ls);
      |  ^~~~~~~~~~~~~~
/tmp/pear/temp/eio/php7/php_eio.c:496:23: error: 'php_eio_cb_t' {aka 'struct <anonymous>'} has no member named 'ls'
  496 |  TSRMLS_SET_CTX(eio_cb->ls);
      |                       ^~
/tmp/pear/temp/eio/php7/php_eio.c: In function 'php_eio_new_eio_cb_custom':
/tmp/pear/temp/eio/php7/php_eio.c:550:23: error: 'php_eio_cb_custom_t' {aka 'struct <anonymous>'} has no member named 'ls'
  550 |  TSRMLS_SET_CTX(eio_cb->ls);
      |                       ^~
make: *** [Makefile:210: php7/php_eio.lo] Error 1
ERROR: `make' failed

**Comment by alexmorbo** [comment]
Any news?

**Comment by rosmanov** [comment]
> Any news?

I can't find the time yet.

**Comment by rosmanov** [comment]
Merged and uploaded version 3.0.0RC1 (beta) to PECL.

There are still things that might be done better. For instance, the callbacks might be embeedded into the eio custom structures through the `zend_fcall_info` and `zend_fcall_info_cache` and we might try to find a way to support ZTS in `eio_custom`. Unfortunately I don't have enough time for this. However, the ext shouldn't be worse than it was at least =)

**Comment by rosmanov** [comment]
Oops, the package uploaded to PECL is broken. Will be fixed asap.

**Comment by rosmanov** [comment]
The PECL package fixed in 3.0.0RC2.

**Comment by enumag** [comment]
Thank you!
